R/PostDefinition.R

```diff
@@ -62,12 +62,34 @@ postDefinition <- function(baseUrl, name, category, definition) {
   } else {
     expression <- definition
   }
-  # convert R-object to JSON expression.
-  jsonExpression <- RJSONIO::toJSON(expression)
-  # create json body
-  json <- paste0("{\"name\":\"", as.character(name), "\",\"expressionType\": \"SIMPLE_EXPRESSION\", \"expression\":",
-                 jsonExpression,
-                 "}")
+  
+  categoryMetaData <- getDefinitionsMetadata(baseUrl = baseUrl, category = category)
+  while(name %in% categoryMetaData$name){
+    name <- paste0(name,"(1)")
+  }
+  
+  expression$createdDate <- NULL
```

Comment:
We need to design this to support all the atlas categories (i.e. characterization, estimation, cohort, concept set). Each of these have metadata like this createDate, modifiedDate etc.. but they are not consistent (esp characterization i think). So i used [this function](https://github.com/OHDSI/ROhdsiWebApi/blob/3e09d7350984c4ec9b4a93678b3492b5fcbf67ef/R/Private.R#L71) to change the returned object used [here ](https://github.com/OHDSI/ROhdsiWebApi/blob/3e09d7350984c4ec9b4a93678b3492b5fcbf67ef/R/GetDefinitionsMetadata.R#L71)

What you are doing is the opposite of what i did - trying to push the webapi, but the name should be change based on whether it is characterization or pathway or estimation etc. See 
[Characterization](http://webapidoc.ohdsi.org/index.html#1410315086)
![image](https://user-images.githubusercontent.com/13936600/86512986-7f056d80-bdd4-11ea-977b-36da1703873a.png)

[cohort-definition](http://webapidoc.ohdsi.org/index.html#398821407)
![image](https://user-images.githubusercontent.com/13936600/86513003-abb98500-bdd4-11ea-91da-2bb3c09353fc.png)

So, this PR needs to figure out how to change names based on what category is being posted. You could modify this [function](https://github.com/OHDSI/ROhdsiWebApi/blob/fe0a0d2978a891df4c8b23a639ec51c50f54c4f7/R/Private.R#L71) use the function like in [here](https://github.com/OHDSI/ROhdsiWebApi/blob/fe0a0d2978a891df4c8b23a639ec51c50f54c4f7/R/PostDefinition.R#L87)

Reply:
Sorry - in retrospect the code I added here was lazy. I'll reverse engineer the existing function.

Reply:
When the correct pathways expression is used, forcing ```NULL``` values is not necessary - removed.